provider/testdata/Cargo.toml

```diff
@@ -93,12 +104,17 @@ locales = [
 
     # Thai:
     # - Complex word breaking
+    # - Ignores punctuation in collation by default
     "th",
 
     # Turkish:
     # - Interesting case-mappings
     "tr",
 
+    # Vietnamese:
+    # - Tailors the collation order of diacritics
+    "vi",
```

Comment:
To add locales selectively for individual components, add them to the `cldr_json_glob` below.

Reply:
This looks like it only works from CLDR JSON data. Is it supposed to work for other data?

Reply:
To run a test using a locale that is not one of the test locales, if the source toml data is present, you can run your test directly off the source data toml.

Reply:
Clarification: `cldr_json_glob` only affects which files are automatically downloaded from CLDR and saved in `testdata/data/cldr`. If your data comes from elsewhere, you don't need to update` cldr_json_glob`.

Reply:
It seems that a specialized collator-specific glob would be useful.

Reply:
The `cldr_json_glob` is only used by the `icu4x-testdata-download` script to limit the amount of data we keep in `provider/testdata/data/cldr`. As we curate `provider/testdata/data/uprops` manually, we don't have or need such a glob for uprops/collator stuff.

Reply:
AFAICT, the allow-list here also limits what goes into the collator test data, so I needed to add items here despite manually curating the test data. (I.e. what goes into the postcard is an intersection of what's manually curated to be available and what is allow-listed here.)

Reply:
That is correct. We definitely do not want to extend the locale set for all test data though. There are two options to test e.g. Vietnamese collation:
* In your crate you can manually construct the `CollationDataV1` with arbitrary data. This lets you test abstract concepts decoupled from actual data ([example](https://github.com/unicode-org/icu4x/blob/main/components/list/src/provider.rs#L269))
* The data transformer (`CollationDataDataProvider`) is backed by all the data in `provider/testdata/data/uprops` and not intersected with this list of locales. There you can test behaviour on specific data. This does not go through serialization/deserialization, but that does not need to be tested. ([example](https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/cldr/transform/list/mod.rs#L246))

Reply:
It seems to me that it would be useful to separate what gets intersected by the test data locale filter from what gets intersected with a user-chosen explicit locale filter. I think it makes sense to do that separately from this PR.

Reply:
This should not be merged like this. These are the explicit locales we're choosing for testing and they should not be extended to cover one more test case for one key.

Reply:
Do you have advice on how to change the datagen to exempt either collation data or all non-CLDR provider data from being intersected with the locale list when the locale list comes from the list of test locales but not when the locale list is provided by the user running datagen for generating non-test data?

Overall, it seems to me it would be more efficient if someone already more familiar with datagen than me could make this change.

Reply:
AFAICT, `IterableResourceProvider` should be able to declare whether it wants to opt out of locale filtering when building test data, so that `RequestFilterDataProvider` could use this declaration to turn into a no-op when the delegate requests opt-out and the creator of the filter indicates that the filter is being created for test data purposes. However, it seems to me that `Filterable` is so type-unconstrained that it doesn't make sense for me to spend a lot of time changing these things without guidance in a way that wouldn't then pass review.

Reply:
Testdata generation uses the standard datagen API, and I don't want to add a `additionally_include_fi_ko_lt_sv_vi_zh_for_collation_data_v1` flag to that.

Which tests require this data, and why can they not be moved into datagen?

I'm available on slack if you want to sync about this.

Reply:
I reverted this file and commented out the tests that depend on languages from outside the `testdata` set.